src/tests.rs

```diff
@@ -0,0 +1,272 @@
+use std::vec;
+
+use hex::FromHex;
+
+use crate::{hash::Hash, PreviousErr, Ratchet};
+
+fn hash_from_hex(s: &str) -> Hash {
+    Hash::from_raw(<[u8; 32]>::from_hex(s).unwrap())
+}
+
+const SEED: &str = "600b56e66b7d12e08fd58544d7c811db0063d7aa467a1f6be39990fed0ca5b33";
+
+#[test]
+fn test_ratchet_zero() {
+    // Seed pulled from https://whitepaper.fission.codes/file-system/partitions/private-directories/concepts/spiral-ratchet
+    let seed = hash_from_hex(SEED);
+
+    let a = &mut Ratchet::zero(seed.into());
+    let expect = &Ratchet {
+        large: hash_from_hex("5aa00b14dd50887cdc0b0b55aa2da1eb5cc3a79cdbe893b2319da378a83ad0c5"),
+        medium: hash_from_hex("5a86c2477e2ae4ffcf6373cce82259eb542b72a72db9cf9cddfe06bcc20623b6"),
+        small: hash_from_hex("962b7f9ac204ffd0fa398e9c875c90806c0cd6646655f7a5994b7a828b70c0da"),
+        medium_counter: 0,
+        small_counter: 0,
+    };
+
+    assert_ratchet_equal(expect, a);
+
+    a.inc();
+
+    let b = &mut Ratchet::zero(seed.into());
+    b.inc();
+
+    assert_ratchet_equal(a, b);
+
+    let a_key = a.key();
+    let b_key = b.key();
+    assert_eq!(a_key, b_key);
+}
+
+#[test]
+fn test_ratchet_add_256() {
+    let seed = hash_from_hex(SEED);
+    // manually advance ratchet 256 (2 ^ 8) times
+    let slow = &mut Ratchet::zero(seed.into());
+    for _ in 0..256 {
+        slow.inc();
+    }
+
+    // fast jump 256 values in one shot
+    let (ref fast, _) = Ratchet::zero(seed.into()).next_medium_epoch();
+    assert_ratchet_equal(slow, fast);
+}
+
+// #[test]
+// fn test_fuzz_ratchet {
+//     // TODO: research rust fuzz impl
```

Comment:
For my IBF stuff I've recently used `proptest`. The docs aren't *amazing*, but it's good to use, and actively updated (unlike quickcheck). Unfortunately you have to use a `proptest!` macro, and I always find the domain-specific macros the hardest thing to learn & you sometimes lose IDE intellisense, but once I got what it's doing it was fine :)

Reply:
Oh. The comment is from the original code. I'm leaving it there cos we do need fuzz testing. I'm checking out proptest.